### DISCUSSION TITLE
Compatibility with Enhancd?

### DISCUSSION THREAD

**Discussion by JP-Ellis** [marlonrichert/zsh-autocomplete#438] [Q&A] [answered]
Until recently, I've been using `zsh-autocomplete` alongside `enhancd`, but a recent update has lead them to become incompatible with each other.

I was curious to know whether anyone else has used both these plugins side-by-side with any success?  Alternatively, is it possible to somehow override how `zsh-autocomplete` completes a particular function?

I am using the latest commits from each repository at the time of writing this.

Here is a screenshot of the autocompletions without `enhancd` (i.e. regular `cd`):

![image](https://user-images.githubusercontent.com/3196162/170143800-9ef8a29c-ae75-469d-824f-ec74d9dba07f.png)

And here is a screenshot of what happens with `enhancd`:

![image](https://user-images.githubusercontent.com/3196162/170144082-6b62a97e-1531-4670-b17c-75f4a9910f49.png)

and with the `enhancd` version, typing `cd /` for example does not display any completions, and I am unable to generate new completions with `<TAB>`.

I should also mention that if I disable `zsh-autocomplete`, `enhancd` works fine again.  So it appears to be specifically the combination of `zsh-autocomplete` and `enhancd` which is causing an issue.  While the screenshot above has a couple of other zsh plugins, the issue is replicated if I disable all other plugins other than `enhancd` and `zsh-autocomplete`.

**Comment by marlonrichert** [comment]
Incompatibility in what way?

**Reply by JP-Ellis** [reply]
Sorry, my bad for not elaborating.  I've updated the original question with examples of what happens.

**Reply by marlonrichert** [reply]
I'm unable to reproduce it:
```zsh
% cd $(mktemp -d)
% git clone --depth 1 -- https://github.com/marlonrichert/zsh-autocomplete.git
Cloning into 'zsh-autocomplete'...
remote: Enumerating objects: 59, done.
remote: Counting objects: 100% (59/59), done.
remote: Compressing objects: 100% (55/55), done.
remote: Total 59 (delta 0), reused 27 (delta 0), pack-reused 0
Receiving objects: 100% (59/59), 1.64 MiB | 2.36 MiB/s, done.
% git clone --depth 1 -- https://github.com/b4b4r07/enhancd.git
Cloning into 'enhancd'...
remote: Enumerating objects: 89, done.
remote: Counting objects: 100% (89/89), done.
remote: Compressing objects: 100% (85/85), done.
remote: Total 89 (delta 3), reused 30 (delta 1), pack-reused 0
Receiving objects: 100% (89/89), 28.02 KiB | 478.00 KiB/s, done.
Resolving deltas: 100% (3/3), done.
% > .zshrc <<EOF
PS1='%# ' PS2= RPS2='%^'; setopt transientrprompt interactivecomments
source ~/zsh-autocomplete/zsh-autocomplete.plugin.zsh
source ~/enhancd/init.sh
EOF
% env -i HOME=$PWD PATH=$PATH TERM=$TERM ${TERMINFO:+TERMINFO=$TERMINFO} zsh -d
% cd
directory
enhancd/           zsh-autocomplete/
all matches
enhancd zsh-autocomplete
```

**Reply by JP-Ellis** [reply]
Interesting, thanks for the minimal example. I was able to reproduce your results which therefore indicates that there's a difference somewhere else, which is weird since I did remove most (though not all) of my default zsh config when trying to identify this issue.

I'll work on your minimal example and see if I can replicate the issue and I'll let you know.

**Comment by ztNFny** [comment]
I'm having the same issue. zsh-autocomplete + enhancd = no completion of filenames with the "cd" command but working fine with other commands (e.g. "ls").
Did you ever manage to identify the problem?

**Comment by ztNFny** [comment]
I'm able to reliably repoduce the issue with this .zshrc:
```
source "$HOME/.zinit/bin/zinit.zsh"
autoload -Uz _zinit
(( ${+_comps} )) && _comps[zinit]=_zinit
zinit load marlonrichert/zsh-autocomplete
zinit load b4b4r07/enhancd
```
All 3 plugins are using todays main branch. But the issue has been existing for weeks now

**Comment by marlonrichert** [comment, marked as answer]
I have changed how to connect dir jumping tools to Autocomplete. Please see the [Readme](https://github.com/marlonrichert/zsh-autocomplete#use-a-custom-backend-for-recent-directories) for more info.
